service/history/shard/context_impl.go

```diff
@@ -849,6 +850,11 @@ func (s *ContextImpl) DeleteWorkflowExecution(
 		}
 	}
 
+	// only allow deleting open workflow in passive clusters
+	if startTime != nil && namespaceEntry.ActiveClusterName() == s.clusterMetadata.GetCurrentClusterName() {
+		return serviceerror.NewInvalidArgument("Cannot delete open workflow in active cluster")
+	}
```

Comment:
It took me some time to recall the relationship between startTime field and workflow is running or not. For people have less context, I guess they will have a hard time figuring this out. Is it possible to move this check to delete manager?